### PR TITLE
Improve update stack

### PR DESCRIPTION
**Pull request by poupotte**
* Add options to block monitor update
 * Update controller only if a new version is available

### CODE REVIEW

**Comment by frankrousseau** [comment]
@nono can you review that one too?

File: server/controllers/applications.coffee [hunk 1]
```diff
@@ -21,26 +22,34 @@ sendError = (res, err, code=500) ->
...
-        if err or stderr
-            if count < 2
-                updateController count + 1, callback
-            else
-                callback "Error during controller update after #{count + 1} try: #{stderr}"
+    # Check if a new version is available
+    latest 'cozy-controller', (err, version) ->
+        if version isnt pkg.version
+            log.info "controller: update"
+            exec "npm -g update cozy-controller", (err, stdout, stderr) ->
```
Comment (nono) [inline server/controllers/applications.coffee]: @poupotte why not using [async.retry](https://github.com/caolan/async#retry) ?

File: server/controllers/applications.coffee [hunk 2]
```diff
@@ -21,26 +22,34 @@ sendError = (res, err, code=500) ->
...
 updateController = (count, callback) ->
-    log.info "controller: update"
-    exec "npm -g update cozy-controller", (err, stdout, stderr) ->
-        if err or stderr
-            if count < 2
-                updateController count + 1, callback
-            else
-                callback "Error during controller update after #{count + 1} try: #{stderr}"
+    # Check if a new version is available
+    latest 'cozy-controller', (err, version) ->
```
Comment (nono) [inline server/controllers/applications.coffee]: @poupotte I think we should check `err` here. If we have an error while fetching the latest version from npm, it's probably not a good time to try updating packages with npm.

**Comment by nono** [comment]
@frankrousseau done, waiting for @poupotte responses

**Comment by nono** [comment]
@poupotte :+1:

**Pull request merged**